style.md

```diff
@@ -1923,6 +1924,112 @@ sptr := &T{Name: "bar"}
 </td></tr>
 </tbody></table>
 
+### Initializing Maps and Slices
+
+Where reasonable, prefer to initialize maps and slices with `make()` versus type
+literals, as the intent is generally easier to understand.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+var (
+  m1 = map[T1]T2{}
+  m2 map[T1]T2
+)
+```
+
+```go
+var (
+  s1 = []T{}
+  s2 []T
+)
+```
+
+</td><td>
+
+```go
+var (
+  m1 = make(map[T1]T2)
+  m2 map[T1]T2
+)
+```
+
+```go
+var (
+  s1 = make([]T, 0)
+  s2 []T
+)
+```
+
+</td></tr>
+<tr><td>
+
+In the above examples, both declaration and initialization are visually similar
+(have a lesser Levenshtein distance).
+
+</td><td>
+
+In the above examples, declaration and initialization are more visually distinct
+(have a greater Levenshtein distance).
+
+</td></tr>
+</tbody></table>
+
+Additionally, given that the signature for `make()` with a map is
+`make(map[T1]T2[, hint])`, refactors that add/remove capacity hints are simpler:
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```diff
+diff --git a/tmp.go b/tmp.go
+index e014e4f64..e07a845d9 100644
+--- a/tmp.go
++++ b/tmp.go
+@@ -1 +1 @@
+-m := map[T1]T2{}
++m := make(map[T1]T2, hint)
+```
+
+</td><td>
+
+```diff
+diff --git a/tmp.go b/tmp.go
+index 8aa33cc41..e07a845d9 100644
+--- a/tmp.go
++++ b/tmp.go
+@@ -1 +1 @@
+-m := make(map[T1]T2)
++m := make(map[T1]T2, hint)
+```
+
+</td></tr>
+</tbody></table>
+
+While it's impractical to prescribe a universal strategy for map and slice
+initialization, there are some basic rules of thumb for when to use `make()`
+versus type literals:
+
+- Use `make(map[T1]T2)` or `make([]T, cap)` when...
+  - an empty map or slice is being initialized
+  - elements must be added programatically, versus as part of a literal
+  - [maps only] when it is optimal to pre-allocate storage (based on an
```

Comment:
I wouldn't point people to the map implementation. We can just suggest adding a size hint when available.

Reply:
sure, that sounds good - mostly copy/pasted this from the related issue.